bitlist64.go

```diff
@@ -0,0 +1,324 @@
+			return false
+		}
+	}
+
+	return true
+}
+
+// Overlaps returns true if the bitlist contains one of the bits from the provided argument
+// bitlist. This method will panic if bitlists are not the same length.
+func (b *Bitlist64) Overlaps(c *Bitlist64) bool {
+	lenB, lenC := b.Len(), c.Len()
+	if lenB != lenC {
+		panic("bitlists are different lengths")
+	}
+
+	if lenB == 0 || lenC == 0 {
+		return false
+	}
+
+	// To ensure all of the bits in c are not overlapped in b, we iterate over every word, invert b
+	// and xor the word from b and c, then and it against c. If the result is non-zero, then
+	// we can be assured that word in c had bits not overlapped in b.
+	for idx, word := range b.data {
+		if (^word^c.data[idx])&c.data[idx]&allBitsSet != 0 {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Or returns the OR result of the two bitfields (union).
+// This method will panic if the bitlists are not the same length.
+func (b *Bitlist64) Or(c *Bitlist64) *Bitlist64 {
+	if b.Len() != c.Len() {
+		panic("bitlists are different lengths")
+	}
+
+	ret := b.Clone()
+	b.NoAllocOr(c, ret)
+
+	return ret
+}
+
+// NoAllocOr computes the OR result of the two bitfields (union).
+// Result is written into provided variable, so no allocation takes place inside the function.
+// This method will panic if the bitlists are not the same length.
+func (b *Bitlist64) NoAllocOr(c, ret *Bitlist64) {
+	if b.Len() != c.Len() {
+		panic("bitlists are different lengths")
+	}
+
+	for idx, word := range b.data {
+		ret.data[idx] = word | c.data[idx]
+	}
+}
+
+// And returns the AND result of the two bitfields (intersection).
+// This method will panic if the bitlists are not the same length.
+func (b *Bitlist64) And(c *Bitlist64) *Bitlist64 {
+	if b.Len() != c.Len() {
+		panic("bitlists are different lengths")
+	}
+
+	ret := b.Clone()
+	b.NoAllocAnd(c, ret)
+
+	return ret
+}
+
+// NoAllocAnd computes the AND result of the two bitfields (intersection).
+// Result is written into provided variable, so no allocation takes place inside the function.
+// This method will panic if the bitlists are not the same length.
+func (b *Bitlist64) NoAllocAnd(c, ret *Bitlist64) {
+	if b.Len() != c.Len() {
+		panic("bitlists are different lengths")
+	}
+
+	for idx, word := range b.data {
+		ret.data[idx] = word & c.data[idx]
+	}
+}
+
+// Xor returns the XOR result of the two bitfields (symmetric difference).
+// This method will panic if the bitlists are not the same length.
+func (b *Bitlist64) Xor(c *Bitlist64) *Bitlist64 {
+	if b.Len() != c.Len() {
+		panic("bitlists are different lengths")
+	}
+
+	ret := b.Clone()
+	b.NoAllocXor(c, ret)
+
+	return ret
+}
+
+// NoAllocXor returns the XOR result of the two bitfields (symmetric difference).
+// Result is written into provided variable, so no allocation takes place inside the function.
+// This method will panic if the bitlists are not the same length.
+func (b *Bitlist64) NoAllocXor(c, ret *Bitlist64) {
+	if b.Len() != c.Len() {
```

Comment:
you should also check the length of `ret` bitlist

Reply:
Nice suggestion, thank you! Will do now.